engine/docker/docker.go

```diff
@@ -324,18 +324,37 @@ func (e *Docker) create(ctx context.Context, pipelineConfig *spec.PipelineConfig
 		)
 	}
 
+	originalImage := step.Image
+	overriddenImage := originalImage
+
+	// override image registry for internal images
+	// this is short term solution
+	// override if no auth is present
+	if isHosted && (step.Auth == nil || step.Auth.Username == "" || step.Auth.Password == "") {
+		overriddenImage = image.OverrideRegistry(step.Image)
+	}
+
+	selectedImage := overriddenImage
+
 	// automatically pull the latest version of the image if requested
 	// by the process configuration, or if the image is :latest
 	if step.Pull == spec.PullAlways ||
-		(step.Pull == spec.PullDefault && image.IsLatest(step.Image)) {
-		pullerr := e.pullImageWithRetries(ctx, step.Image, pullopts, output)
+		(step.Pull == spec.PullDefault && image.IsLatest(overriddenImage)) {
+		pullerr := e.pullImageWithRetries(ctx, overriddenImage, pullopts, output)
 		if pullerr != nil {
-			return pullerr
+			// if for some reason overridden image does not work then fallback
+			if overriddenImage != originalImage {
+				selectedImage = originalImage
+				pullerr = e.pullImageWithRetries(ctx, originalImage, pullopts, output)
```

Comment:
Should you not be using `selectedImage` here? or remove line 347.

Reply:
selectedImage is used for container creation only because  overriddenImage is re used in case of PullNever or not found cases later